idea/idea-gradle/src/org/jetbrains/kotlin/idea/inspections/gradle/KotlinGradleModelFacade.kt

```diff
@@ -66,11 +64,7 @@ fun getDependencyModules(moduleData: DataNode<ModuleData>, gradleIdeaProject: Id
     return emptyList()
 }
 
-fun findModulesByNames(dependencyModuleNames: List<String>, gradleIdeaProject: IdeaProject, ideProject: DataNode<ProjectData>): LinkedHashSet<DataNode<ModuleData>> {
-    return dependencyModuleNames.mapNotNullTo(LinkedHashSet()) { targetModuleName ->
-        val targetGradleModule = gradleIdeaProject.modules.firstOrNull { it.name == targetModuleName } ?: return@mapNotNullTo null
-        ExternalSystemApiUtil.findFirstRecursively(ideProject) {
-            (it.data as? ModuleData)?.id == targetGradleModule.gradleProject.path
-        } as DataNode<ModuleData>?
-    }
+fun findModulesByNames(dependencyModuleNames: Set<String>, gradleIdeaProject: IdeaProject, ideProject: DataNode<ProjectData>): LinkedHashSet<DataNode<ModuleData>> {
+    val modules = ExternalSystemApiUtil.getChildren(ideProject, ProjectKeys.MODULE)
+    return modules.filterTo(LinkedHashSet()) { it.data.externalName in dependencyModuleNames }
```

Comment:
Why we change recursive search to search among children? Do I miss something?

Reply:
Modules are stored as children of Project node. Full recursive search will visit huge number of extra nodes (e.g., dependenices).